src/lucky_record/errors.cr

```diff
@@ -0,0 +1,20 @@
+# = Lucky Record Errors
+#
+# Generic Lucky Record exception class.
+class LuckyRecordError < Exception
+end
+
+# Raised when Lucky Record cannot find a record by given id
+class RecordNotFound < LuckyRecordError
+  getter model : (String|Nil)
+  getter primary_key : (String|Nil)
+  getter id : (String|Int32|Nil)
+
+  def initialize(message : (String|Nil) = nil, model : (String|Nil) = nil, primary_key : (String|Nil) = nil, id : (String|Int32|Nil) = nil)
+    @primary_key = primary_key
+    @model = model
+    @id = id
+
+    super(message)
```

Comment:
Might be cool to make a custom message like `Could not find #{model} with id of #{id}`

You can see here that I did that in Habitat by calling super with a string: https://github.com/luckyframework/habitat/blob/master/src/habitat.cr#L4

So this would look like `super "Could not find #{model} with id of #{id}"`